linter/lints/cabf_br/lint_crl_validity_period.go

```diff
@@ -42,17 +47,73 @@ func (l *crlValidityPeriod) CheckApplies(c *x509.RevocationList) bool {
 }
 
 func (l *crlValidityPeriod) Execute(c *x509.RevocationList) *lint.LintResult {
-	validity := c.NextUpdate.Sub(c.ThisUpdate)
-	if validity <= 0 {
+	/*
+		Let's Encrypt issues CRLs for two distinct purposes:
+		   1) CRLs containing subscriber certificates created by the
+		      crl-updater. These CRLs must have only the distributionPoint and
+		      onlyContainsUserCerts fields set.
+		   2) CRLs containing subordinate CA certificates created by the
+		      ceremony tool. These CRLs must only have the onlyContainsCACerts
+		      field set.
+
+		RFC 5280 Section 5.2.5
+
+		IssuingDistributionPoint ::= SEQUENCE {
+			distributionPoint          [0] DistributionPointName OPTIONAL,
+			onlyContainsUserCerts      [1] BOOLEAN DEFAULT FALSE,
+			onlyContainsCACerts        [2] BOOLEAN DEFAULT FALSE,
+			...
+		}
+	*/
+
+	// The only way to determine which type of CRL we're dealing with, the
+	// issuingDistributionPoint must be parsed and the internal fields
+	// inspected.
+	idpOID := asn1.ObjectIdentifier{2, 5, 29, 28} // id-ce-issuingDistributionPoint
+	idpe := lints.GetExtWithOID(c.Extensions, idpOID)
+	if idpe == nil {
+		return &lint.LintResult{
+			Status:  lint.Warn,
+			Details: "CRL missing IDP",
+		}
+	}
+
+	// Step inside the outer issuingDistributionPoint sequence to get access to
+	// its constituent fields.
+	idpv := cryptobyte.String(idpe.Value)
+	_ = idpv.ReadASN1(&idpv, cryptobyte_asn1.SEQUENCE)
+
```

Comment:
It would be great if this lint were slightly more robust. Right now, a CRL with both a `distributionPoint` and the `onlyContainsCACerts` boolean<sup>1</sup> will be treated as though it were a `onlyContainsUserCerts` CRL. This is because this cryptobyte code only modifies the BRValidity variable if the _first_ tag encountered within the extension is `[2]`.

Instead, I think it would be better for this lint to blindly throw away any bytes with tag `[0]` before looking for the `onlyContainsCACerts` field. You can do this with `idpv.SkipOptionalASN1(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed())`.

[1]: To be clear, we don't currently issue any certs that look like this, but we _could_ in the future.

Reply:
I've created a test case for this, thanks!